upf/upf_forward.c

```diff
@@ -213,8 +214,16 @@ upf_forward (vlib_main_t * vm, vlib_node_runtime_t * node,
 				       far->forward.fp_pool_index);
 		  vnet_buffer (b)->ip.adj_index[VLIB_TX] =
 		    fp_entry->dpo.dpoi_index;
+
+		  vec_foreach (rpath, fp_entry->rpaths)
+		  {
+		    if (is_ip4 && rpath->frp_proto == DPO_PROTO_IP4)
+		      break;
+		    if (!is_ip4 && rpath->frp_proto == DPO_PROTO_IP6)
+		      break;
+		  }
 		  vnet_buffer (b)->sw_if_index[VLIB_TX] =
-		    fp_entry->rpaths->frp_fib_index;
+		    rpath->frp_fib_index;
```

Comment:
Why it is fib_index, but assigned to `sw_if_index` value? From what I understand, interfaces have their own indexing that does not directly correspond to FIB indices

Reply:
Ok I see:
https://github.com/FDio/vpp/blob/a424dd1/src/vnet/ip/ip4_forward.c#L80-L86